dask_expr/_backends.py

```diff
@@ -0,0 +1,95 @@
+from __future__ import annotations
+
+import pandas as pd
+from dask.backends import CreationDispatch
+from dask.dataframe.backends import DataFrameBackendEntrypoint
+
+from dask_expr._dispatch import get_collection_type
+
+
+class DXCreationDispatch(CreationDispatch):
+    """Dask-expressions version of CreationDispatch"""
+
+    # TODO Remove after https://github.com/dask/dask/pull/10794
+    def dispatch(self, backend: str):
+        from dask.backends import detect_entrypoints
+
+        try:
+            impl = self._lookup[backend]
+        except KeyError:
+            entrypoints = detect_entrypoints(f"dask_expr.{self._module_name}.backends")
```

Comment:
The only reason we cannot use `CreationDispatch` directly is the fact that `dask_cudf` is already using the `"dask.dataframe.backends"` path to expose the `"cudf"` backend entrypoint for `dask.dataframe`.

We can just use `CreationDispatch` directly after https://github.com/dask/dask/pull/10794 makes it possible to modify the entrypoint path upon initialization.

Reply:
Scratch that - I should probably wait for the next dask/dask release to remove DXCreationDispatch